From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 31 Dec 2025 18:38:11 +0100
Subject: [PATCH] Set soversion

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 61480c0..114cf80 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -966,6 +966,7 @@ target_include_directories(mdbx-static INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}")
 if(MDBX_BUILD_SHARED_LIBRARY)
   add_library(mdbx SHARED ${LIBMDBX_SOURCES})
   set_target_properties(mdbx PROPERTIES PUBLIC_HEADER "${LIBMDBX_PUBLIC_HEADERS}")
+  set_target_properties(mdbx PROPERTIES VERSION "${LIBRARY_VERSION}" SOVERSION "${MDBX_VERSION_MAJOR}.${MDBX_VERSION_MINOR}.${MDBX_VERSION_RELEASE}")
   target_compile_definitions(
     mdbx
     PRIVATE LIBMDBX_EXPORTS MDBX_BUILD_SHARED_LIBRARY=1
